spec/lib/carma/client/client_spec.rb

```diff
@@ -26,22 +28,36 @@
   end
 
   describe '#create_submission' do
-    it 'accepts a payload and submitts to CARMA' do
-      payload           = { 'my' => 'data' }
-      response_double   = double
+    def self.test_carma_submission
+      it 'accepts a payload and submits to CARMA' do
+        payload           = { 'my' => 'data' }
+        response_double   = double
 
-      expect(restforce_client).to receive(:post).with(
-        '/services/apexrest/carma/v1/1010-cg-submissions',
-        payload,
-        'Content-Type': 'application/json',
-        'Sforce-Auto-Assign': 'FALSE'
-      ).and_return(
-        response_double
-      )
+        expect(restforce_client).to receive(:post).with(
+          '/services/apexrest/carma/v1/1010-cg-submissions',
+          payload,
+          'Content-Type': 'application/json',
+          'Sforce-Auto-Assign': 'FALSE'
+        ).and_return(
+          response_double
+        )
 
-      expect(response_double).to receive(:body).and_return(:response_token)
-      response = subject.create_submission(payload)
-      expect(response).to eq(:response_token)
+        expect(response_double).to receive(:body).and_return(:response_token)
+        response = subject.create_submission(payload)
+        expect(response).to eq(:response_token)
+      end
+    end
+
+    context 'with betamocks enabled' do
+      test_carma_submission
+    end
+
+    context 'with betamocks disabled' do
+      before do
+        Settings['salesforce-carma'].mock = false
```

Comment:
I think overriding this global variable could potentially result in unexpected side effects or confusion to a developer.

I'd mock this instead of actually overriding it:
```ruby
expect(Settings['salesforce-carma']).to receive(:mock).and_return(false)
```